### PR TITLE
[FLINK-2305] Add documenation about Storm compatibility layer

### PR DESCRIPTION
**Pull request by mjsax**
additional improvements
- added RawOutputFormatter
- bug fix in SimpleOutputFormatter
- enable default output formatter for StormBoltFileSink

### CODE REVIEW

**Comment by mjsax** [comment]
FYI: this PR is related to #878 -> if #878 is merged before, this PR needs an update.

File: docs/apis/storm_compatibility.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+If a parameter is not specified, the value is taken from `flink-conf.yaml`.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+~~~java
+FlinkTopologyBuilder builder = new FlinkTopologyBuilder(); // replaces: TopologyBuilder builder = new FlinkTopology();
+
+builder.setSpout("source", new StormFileSpout(inputFilePath));
+builder.setBolt("tokenizer", new StormBoltTokenizer()).shuffleGrouping("source");
+builder.setBolt("counter", new StormBoltCounter()).fieldsGroupign("tokenizer", new Fields("word"));
```
Comment (rmetzger) [inline docs/apis/storm_compatibility.md]: typo in `fieldsGroupign`

**Comment by rmetzger** [comment]
+1 to merge!

File: docs/apis/storm_compatibility.md (hunk 1 above)
Comment (mjsax) [inline docs/apis/storm_compatibility.md]: fixed. :)

**Comment by StephanEwen** [comment]
I just tested the storm compatibility code, and stumbled across the following issues in the logs:

1) Multiple SLF4J bindings. If Storm brings its own one, we may need to define an exclusion in the storm dependencies.
```
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/cicero/.m2/repository/ch/qos/logback/logback-classic/1.0.13/logback-classic-1.0.13.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/cicero/.m2/repository/org/slf4j/slf4j-log4j12/1.7.7/slf4j-log4j12-1.7.7.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
```


2) Improper Task life cycle. The logs below indicate that the JobManager and TaskManager are in some way killed/removed before the tasks are properly removed and before the client learns about the JobStatus.

I am a bit puzzled how you actually manage to provoke this ;-)
```
org.apache.flink.runtime.client.JobExecutionException: Communication with JobManager failed: Lost connection to JobManager akka://flink/user/jobmanager
	at org.apache.flink.runtime.client.JobClient.submitJobAndWait(JobClient.java:169)
	at org.apache.flink.runtime.minicluster.FlinkMiniCluster.submitJobAndWait(FlinkMiniCluster.scala:205)
	at org.apache.flink.runtime.minicluster.FlinkMiniCluster.submitJobAndWait(FlinkMiniCluster.scala:195)
	at org.apache.flink.streaming.util.TestStreamEnvironment$1.run(TestStreamEnvironment.java:116)
Caused by: java.lang.Exception: Lost connection to JobManager akka://flink/user/jobmanager
	at org.apache.flink.runtime.client.JobClientActor.onReceive(JobClientActor.java:131)
	at akka.actor.UntypedActor$$anonfun$receive$1.applyOrElse(UntypedActor.scala:167)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:465)
	at akka.actor.UntypedActor.aroundReceive(UntypedActor.scala:97)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
	at akka.actor.dungeon.DeathWatch$class.receivedTerminated(DeathWatch.scala:46)
	at akka.actor.ActorCell.receivedTerminated(ActorCell.scala:369)
	at akka.actor.ActorCell.autoReceiveMessage(ActorCell.scala:501)
	at akka.actor.ActorCell.invoke(ActorCell.scala:486)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:254)
	at akka.dispatch.Mailbox.run(Mailbox.scala:221)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:231)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 16.294 sec - in org.apache.flink.stormcompatibility.exclamation.StormExclamationLocalITCase
21:40:16.300 [exclamation1 (1/3)] ERROR org.apache.flink.runtime.taskmanager.Task  - FATAL - exception in task resource cleanup
java.lang.IllegalStateException: Memory manager has been shut down.
	at org.apache.flink.runtime.memorymanager.DefaultMemoryManager.releaseAll(DefaultMemoryManager.java:396) ~[flink-runtime-0.10-SNAPSHOT.jar:0.10-SNAPSHOT]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:653) ~[flink-runtime-0.10-SNAPSHOT.jar:0.10-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_25]
```

**Comment by mjsax** [comment]
1) Agreed. Should I add this to this PR or open a JIRA and a new PR?
2) This is a know issues that will be resolved with the new STOP signal.

**Comment by StephanEwen** [comment]
Let's make it a separate issue.

If there is a followup on these two issues, we can merge this, in my opinion.

**Comment by mjsax** [comment]
I just created two JIRAs:
- SLF4J: https://issues.apache.org/jira/browse/FLINK-2337
- JobExecutionException: https://issues.apache.org/jira/browse/FLINK-2338

This PR should be ready to get merged.

**Comment by StephanEwen** [comment]
+1 good to go, with the followup issues being addresses in separate issues.

Merging this...

**Comment by StephanEwen** [comment]
Ah, just saw it is already merged. Github is out of sync...

**Comment by mjsax** [comment]
@gyfora merged this already! PR is not closed because Github is behind aft-git!

**Pull request closed without merging**